sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -226,11 +235,113 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
+            return
+        try:
+            async with self._update_lock:
+                for (key, label), etag in self._refresh_on.items():
+                    updated_sentinel = await self._client.get_configuration_setting(
+                        key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, **kwargs
+                    )
+                    if updated_sentinel is not None:
+                        logging.debug(
+                            "Refresh all triggered by key: %s label %s.",
+                            key,
+                            label,
+                        )
+                        await self._load_all(**kwargs)
+                        self._refresh_on[(key, label)] = updated_sentinel.etag
+                        self._refresh_timer.reset()
+                        return
+        except (ServiceRequestError, ServiceResponseError) as e:
+            logging.debug("Failed to refresh, retrying: %r", e)
+            self._refresh_timer.retry()
+        except HttpResponseError as e:
+            # If we get an error we should retry sooner than the next refresh interval
+            self._refresh_timer.retry()
+            if _is_retryable_error(e):
+                return
+            if self._on_refresh_error:
+                await self._on_refresh_error(e)
+                return
+            raise
+        except Exception as e:
+            if self._on_refresh_error:
+                await self._on_refresh_error(e)
+                return
+            raise
+
+    async def _load_all(self, **kwargs):
+        configuration_settings = {}
+        for select in self._selects:
+            configurations = self._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            async for config in configurations:
+                key = self._process_key_name(config)
+                value = self._process_key_value(config)
+
+                if isinstance(config, FeatureFlagConfigurationSetting):
+                    feature_management = configuration_settings.get(FEATURE_MANAGEMENT_KEY, {})
+                    feature_management[key] = value
+                    if FEATURE_MANAGEMENT_KEY not in configuration_settings:
+                        configuration_settings[FEATURE_MANAGEMENT_KEY] = feature_management
+                else:
+                    configuration_settings[key] = value
+                # Every time we run load_all, we should update the etag of our refresh sentinels
+                # so they stay up-to-date.
+                # Sentinel keys will have unprocessed key names, so we need to use the original key.
+                if (config.key, config.label) in self._refresh_on:
+                    self._refresh_on[(config.key, config.label)] = config.etag
+        self._dict = configuration_settings
+
+    def _process_key_name(self, config):
```

Comment:
Doesn't look like this code needed to be duplicated from the sync version?

Reply:
This is a class method at the moment and the Async class doesn't extend the sync version.

Reply:
Well it could be moved out to a standalone function - but not a big deal as it's pretty minor duplication.